examples/map/map_segment.py

```diff
@@ -0,0 +1,118 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+
+A demostration creating a segment of a particular map from
+transformed coordinates.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+from reproject import reproject_interp
+
+import astropy.units as u
+from astropy.wcs import WCS
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import Helioprojective, NorthOffsetFrame, RotatedSunFrame, transform_with_sun_center
+from sunpy.data.sample import AIA_171_ROLL_IMAGE
+
+###############################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_ROLL_IMAGE)
+
+###############################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel. From this we then transform the coordinates to the required frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+###############################################################################
+# Let's then segment the data based on coordinates and convert them to pixel
+# index. We convert the segmented pixel values to 0.
+segment = np.logical_and(all_hgs.lon < 10 * u.deg, all_hgs.lon > 0 * u.deg)
+
+###############################################################################
+# Let's plot the segment separately
+new_data = np.asarray(segment, dtype='float')
+new_data = np.where(new_data == False, np.nan, new_data)
+
+new_frame_map = sunpy.map.Map(new_data * smap.data, smap.meta)
```

Comment:
If you want to mask out the other bits of the map, I think setting `map.mask` would be a better pattern.

I think this would work:
```suggestion
smap.mask = segment
smap.plot()
```

Reply:
This change would basically remove the segment from `smap`, I was thinking of plotting the segment separately.

Reply:
You mean it would mask out the segment rather than masking out the rest of the image?

Reply:
yeah, wouldn't that be more appropriate for the example?

Reply:
yeah that makes sense, in that case run the mask through `np.logical_not` first to reverse it.

Reply:
I have added this